ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs

```diff
@@ -0,0 +1,161 @@
+                        // Read from file store.
+                        let blob_file =
+                            Object::download(&bucket_name, &generate_blob_name(current_version))
+                                .await
+                                .expect("[indexer gcs] Failed to get file store metadata.");
+                        let blob: TransactionsBlob = serde_json::from_slice(&blob_file)
+                            .expect("[indexer gcs] Failed to deserialize blob.");
+                        blob.transactions
+                    },
+
+                    CacheCoverageStatus::DataNotReady => {
+                        sleep(Duration::from_millis(DATA_NOT_READY_SLEEP_DURATION));
+                        continue;
+                    },
+                    CacheCoverageStatus::CacheHit => {
+                        get_cache_transactions(&mut conn, current_version)
+                            .await
+                            .expect("[Indexer Data] Failed to get cache transactions.")
+                    },
+                };
+
+                let item = RawDatastreamResponse {
+                    response: Some(DatastreamProtoResponse::Data(TransactionsOutput {
+                        transactions: encoded_proto_data_vec
+                            .iter()
+                            .enumerate()
+                            .map(|(i, e)| TransactionOutput {
+                                encoded_proto_data: e.clone(),
+                                version: current_version + i as u64,
+                                ..TransactionOutput::default()
+                            })
+                            .collect(),
+                    })),
+                    chain_id,
+                };
+                match tx.try_send(Result::<_, Status>::Ok(item.clone())) {
+                    Ok(_) => {},
+                    Err(TrySendError::Full(_)) => {
+                        warn!(
+                            request_id = request_id.as_str(),
+                            "[Indexer Data] Receiver is full; retrying."
+                        );
+                        std::thread::sleep(Duration::from_secs(1));
+                        continue;
+                    },
+                    Err(TrySendError::Closed(_)) => {
+                        warn!(
+                            request_id = request_id.as_str(),
+                            "[Indexer Data] Receiver is closed; exiting."
+                        );
+                        break;
+                    },
+                }
+                current_version += BLOB_TRANSACTION_CHUNK_SIZE;
+                ma.tick_now(BLOB_TRANSACTION_CHUNK_SIZE);
+                info!(
+                    request_id = request_id.as_str(),
+                    current_version = current_version,
+                    batch_size = BLOB_TRANSACTION_CHUNK_SIZE,
+                    tps = (ma.avg() * 1000.0) as u64,
```

Comment:
This is the chunk size right?

Reply:
yea, `1000` is the batch size for grpc stream; the moving average window size is the sliding window size, not related to actual logic.